### PR TITLE
Add OTLP ingest feature

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Add `otlp` feature to configure [OTLP ingest](https://docs.datadoghq.com/tracing/trace_collection/open_standards/otlp_ingest_in_the_agent/?tab=host).

### Motivation

Allow users to configure this feature via the Datadog operator.

### Additional Notes

Configuration has the same schema as DataDog/helm-charts/pull/619. This is slightly different from the Agent's, since it has the `enabled` flags explicitly.

As it happens in the Helm chart, this PR only adds support for defining an OTLP/gRPC endpoint with a host and port, and not via a Unix socket.

### Describe your test plan

Test the Datadog operator with the following test cases:

- [x] Setting the gRPC endpoint to `unix:/path/to/socket.sock` results in an error
- [x] Setting the gRPC endpoint to `unix-abstract:opaquedata` results in an error
- [x] Setting the gRPC endpoint to `0.0.0.0` results in an error 
- [x] Setting the HTTP endpoint to `0.0.0.0` results in an error
- [x] Setting both endpoints with an explicit port works correctly.
- [x] Only setting gRPC endpoint works correctly.
- [x] Only setting HTTP endpoint works correctly.
- [x] Setting the gRPC section with enabled:true and no endpoint works with the default value.
- [x] Setting the HTTP section with enabled:true and no endpoint works with the default value.
- [x] Setting both sections with enabled:true and no endpoint works with the default value.
- [x] Not setting anything related to OTLP works correctly.

### CODE REVIEW

File: docs/configuration.v1alpha1.md [hunk 1]
```diff
@@ -213,6 +213,10 @@ spec:
 | agent.networkPolicy.create | If true, create a NetworkPolicy for the current agent. |
 | agent.networkPolicy.dnsSelectorEndpoints | Cilium selector of the DNS server entity. |
 | agent.networkPolicy.flavor | Which network policy to use. Can be `kubernetes` or `cilium`. |
+| agent.otlp.receiver.protocols.grpc.enabled | Enable the OTLP/gRPC endpoint. |
+| agent.otlp.receiver.protocols.grpc.endpoint | Endpoint for OTLP/gRPC. gRPC supports several naming schemes: https://github.com/grpc/grpc/blob/master/doc/naming.md The Datadog Operator only supports 'host:port' (usually '0.0.0.0:port'). The default value is '0.0.0.0:4317'. |
```
Comment (kayayarai) [inline docs/configuration.v1alpha1.md:217]:
```suggestion
| agent.otlp.receiver.protocols.grpc.endpoint | Endpoint for OTLP/gRPC. gRPC supports several naming schemes: https://github.com/grpc/grpc/blob/master/doc/naming.md The Datadog Operator supports only `host:port` (usually `0.0.0.0:port`). Default: `0.0.0.0:4317` |
```

File: docs/configuration.v1alpha1.md [hunk 2]
```diff
@@ -213,6 +213,10 @@ spec:
...
+| agent.otlp.receiver.protocols.http.enabled | Enable the OTLP/HTTP endpoint. |
+| agent.otlp.receiver.protocols.http.endpoint | Endpoint for OTLP/HTTP. The default value is '0.0.0.0:4318'. |
```
Comment (kayayarai) [inline docs/configuration.v1alpha1.md:219]:
```suggestion
| agent.otlp.receiver.protocols.http.endpoint | Endpoint for OTLP/HTTP. Default: `0.0.0.0:4318` |
```

File: docs/configuration.v2alpha1.md [hunk 3]
```diff
@@ -99,6 +99,10 @@ spec:
 | features.orchestratorExplorer.enabled | Enabled enables the Orchestrator Explorer. Default: true |
 | features.orchestratorExplorer.extraTags | Additional tags to associate with the collected data in the form of `a b c`. This is a Cluster Agent option distinct from DD_TAGS that is used in the Orchestrator Explorer. |
 | features.orchestratorExplorer.scrubContainers | ScrubContainers enables scrubbing of sensitive container data (passwords, tokens, etc. ). Default: true |
+| features.otlp.receiver.protocols.grpc.enabled | Enable the OTLP/gRPC endpoint. |
+| features.otlp.receiver.protocols.grpc.endpoint | Endpoint for OTLP/gRPC. gRPC supports several naming schemes: https://github.com/grpc/grpc/blob/master/doc/naming.md The Datadog Operator only supports 'host:port' (usually '0.0.0.0:port'). The default value is '0.0.0.0:4317'. |
```
Comment (kayayarai) [inline docs/configuration.v2alpha1.md:103]:
```suggestion
| features.otlp.receiver.protocols.grpc.endpoint | Endpoint for OTLP/gRPC. gRPC supports several naming schemes: https://github.com/grpc/grpc/blob/master/doc/naming.md The Datadog Operator supports only `host:port` (usually `0.0.0.0:port`). Default: `0.0.0.0:4317` |
```

File: docs/configuration.v2alpha1.md [hunk 4]
```diff
@@ -99,6 +99,10 @@ spec:
...
+| features.otlp.receiver.protocols.http.enabled | Enable the OTLP/HTTP endpoint. |
+| features.otlp.receiver.protocols.http.endpoint | Endpoint for OTLP/HTTP. The default value is '0.0.0.0:4318'. |
```
Comment (kayayarai) [inline docs/configuration.v2alpha1.md:105]:
```suggestion
| features.otlp.receiver.protocols.http.endpoint | Endpoint for OTLP/HTTP. Default: `0.0.0.0:4318` |
```

**Review by kayayarai — APPROVED** [review]
A few formatting edits

**Review by celenechang — APPROVED** [review]

**Pull request merged**